lib/hotel_system.rb

```diff
@@ -0,0 +1,49 @@
+
+module Hotel
+  class HotelSystem
+    attr_reader :rooms, :reservations
+    def initialize
+      @rooms = (1..20).to_a 
+      @reservations = []
+    end
+    
+    def list_rooms 
+      return @rooms.dup
+    end
+
+    def available_rooms(date_range)
+      rooms = @rooms.dup
+    end
+
+    def list_reservations(date_range)
+      matches_reservations = @reservations.select do |reservation|
+        reservation.date_range.overlap?(date_range)
```

Comment:
Again, this is a great use of object composition! All your functionality is built into `overlap?` which you can call on this instance of `DateRange`.